xdl_rs/src/conversions/primitive.rs

```diff
@@ -0,0 +1,153 @@
+use crate::xdl_primitive::{XdlPrimitive, XdlPrimitiveMetadata};
+
+impl From<&XdlPrimitive> for XdlPrimitiveMetadata {
+    fn from(x: &XdlPrimitive) -> Self {
+        match x {
+            XdlPrimitive::Bool(_) => XdlPrimitiveMetadata::Bool,
+            XdlPrimitive::U8(_) => XdlPrimitiveMetadata::U8,
+            XdlPrimitive::U16(_) => XdlPrimitiveMetadata::U16,
+            XdlPrimitive::U32(_) => XdlPrimitiveMetadata::U32,
+            XdlPrimitive::U64(_) => XdlPrimitiveMetadata::U64,
+            XdlPrimitive::U128(_) => XdlPrimitiveMetadata::U128,
+            XdlPrimitive::U256(_) => XdlPrimitiveMetadata::U256,
+            XdlPrimitive::I8(_) => XdlPrimitiveMetadata::I8,
+            XdlPrimitive::I16(_) => XdlPrimitiveMetadata::I16,
+            XdlPrimitive::I32(_) => XdlPrimitiveMetadata::I32,
+            XdlPrimitive::I64(_) => XdlPrimitiveMetadata::I64,
+            XdlPrimitive::I128(_) => XdlPrimitiveMetadata::I128,
+            XdlPrimitive::I256(_) => XdlPrimitiveMetadata::I256,
+            XdlPrimitive::F32(_) => XdlPrimitiveMetadata::F32,
+            XdlPrimitive::F64(_) => XdlPrimitiveMetadata::F64,
+            XdlPrimitive::String(_) => XdlPrimitiveMetadata::String,
+        }
+    }
+}
+
+macro_rules! xdl_primitive_from_impl {
+    ($ty:ty, $xdl_type:tt) => {
+        impl From<$ty> for XdlPrimitive {
+            fn from(x: $ty) -> Self {
+                XdlPrimitive::$xdl_type(x)
+            }
+        }
+    };
+}
+
+xdl_primitive_from_impl!(bool, Bool);
+
+xdl_primitive_from_impl!(u8, U8);
+xdl_primitive_from_impl!(u16, U16);
+xdl_primitive_from_impl!(u32, U32);
+xdl_primitive_from_impl!(u64, U64);
+xdl_primitive_from_impl!(u128, U128);
```

Comment:
Sad we can't test 256 yet but given that there's no base type for that currently I'm not sure how we'd pull that off